### PR TITLE
libcurl: add msys2 build_requires for mingw

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **libcurl/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -112,7 +112,10 @@ def system_requirements(self):
         pass
 
     def build_requirements(self):
-        if self._is_win_x_android:
+        if self._is_mingw and "CONAN_BASH_PATH" not in os.environ and \
+           tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+        elif self._is_win_x_android:
             self.build_requires("ninja/1.9.0")
```
Comment (SSE4) [inline recipes/libcurl/all/conanfile.py:119]: not sure why ninja is hard requirement for Android, but it appears to be irrelevant

Comment (SpaceIm) [reply recipes/libcurl/all/conanfile.py:119]: https://github.com/conan-io/conan-center-index/pull/1585#discussion_r426700559

**Comment by SpaceIm** [comment]
It is worth noting that if shared and MinGW, curl.exe fails at link time with a lot of undefined references

File: recipes/libcurl/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/libcurl/all/conanfile.py:119]: that doesn't explain why it needs to use Ninja generator for Android, I believe it's okay to use Make as well, but since it's offtopic, we can discuss in another PR
